node/internal/actors/control_api.go

```diff
@@ -45,11 +47,12 @@ type ControlAPI struct {
 	auctionResponse  auctionResponseFunc
 	nodePingResponse nodePingResponseFunc
 	nodeInfoResponse nodeInfoResponseFunc
+	nodeLameDuck     nodeLameDuckFunc
 
 	self *goakt.PID
 }
 
-func CreateControlAPI(nc *nats.Conn, logger *slog.Logger, publicKey string, aF auctionResponseFunc, nF nodePingResponseFunc, iF nodeInfoResponseFunc) *ControlAPI {
+func CreateControlAPI(nc *nats.Conn, logger *slog.Logger, publicKey string, aF auctionResponseFunc, nF nodePingResponseFunc, iF nodeInfoResponseFunc, ldF nodeLameDuckFunc) *ControlAPI {
```

Comment:
rather than passing in 4 different functions, we should probably extract that into a callback interface and just pass the one thing. This is the pattern being used in the agent common interface.

Reply:
okay....this slowly grew from 1 to 2 to 3 to 4 functions over a week...so good point.  Will look into interface